### ISSUE TITLE
Ensure installations may filter available runtimes to product-only ones

### ISSUE THREAD

**Issue by ALRubinger** [fabric8-launcher/launcher-planning#207]
Raised by the Integreatly team in: https://issues.jboss.org/browse/INTLY-1268?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel

But may be more universally applied.  I know we do have some filtering mechanisms already available, so let's ensure our featureset cleanly allows for this.

Sending to @quintesse as he seemed familiar with this enough when we discussed it :)

**Comment by quintesse** [comment]
This is most likely already possible, I'd have to test it to be sure (such a long time since we've used this feature I'm not 100% sure it's still working or working as I remember). But it would be something like setting this ENV var:

```
export LAUNCHER_BOOSTER_CATALOG_FILTER=booster.version.name.indexOf('redhat')>=0
```

**Comment by gastaldi** [comment]
You can find more examples here: https://github.com/fabric8-launcher/launcher-backend/#filtering-the-booster-catalog

**Comment by gastaldi** [comment]
@quintesse BTW I am afraid that this feature is broken with the `booster-catalog-service` conversion to Kotlin. Here is a sample code that reproduces the issue I am seeing: 
```java
        Predicate<RhoarBooster> filter = BoosterPredicates.withScriptFilter("booster.mission.id == 'rest-http'");
        RhoarBoosterCatalogService service = new RhoarBoosterCatalogService.Builder()
                .filter(filter) // REMOVE THIS LINE AND IT RUNS
                .build();
        service.index().get();
        for (RhoarBooster booster : service.getBoosters()) {
            System.out.println("TESTING " + booster);
            System.out.println(filter.test(booster));
        }
```

Fails with: 
```java
Caused by: <eval>:1 TypeError: Cannot get property "id" of null
	at jdk.nashorn.internal.runtime.ECMAErrors.error(ECMAErrors.java:57)
	at jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:213)
	at jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:185)
	at jdk.nashorn.internal.runtime.ECMAErrors.typeError(ECMAErrors.java:172)
	at jdk.nashorn.internal.runtime.linker.NashornBottomLinker.linkNull(NashornBottomLinker.java:183)
	at jdk.nashorn.internal.runtime.linker.NashornBottomLinker.getGuardedInvocation(NashornBottomLinker.java:66)
	at jdk.internal.dynalink.support.CompositeGuardingDynamicLinker.getGuardedInvocation(CompositeGuardingDynamicLinker.java:124)
	at jdk.internal.dynalink.support.LinkerServicesImpl.getGuardedInvocation(LinkerServicesImpl.java:154)
	at jdk.internal.dynalink.DynamicLinker.relink(DynamicLinker.java:253)
	at jdk.nashorn.internal.scripts.Script$\^eval\_.:program(<eval>:1)
	at jdk.nashorn.internal.runtime.ScriptFunctionData.invoke(ScriptFunctionData.java:637)
	at jdk.nashorn.internal.runtime.ScriptFunction.invoke(ScriptFunction.java:494)
	at jdk.nashorn.internal.runtime.ScriptRuntime.apply(ScriptRuntime.java:393)
	at jdk.nashorn.api.scripting.NashornScriptEngine.evalImpl(NashornScriptEngine.java:421)
```

Remove the `filter()` call and the program runs

**Comment by quintesse** [comment]
@gastaldi that bug has since been fixed (see #96)

**Comment by gastaldi** [comment]
Implemented, closing
